pkg/registry/batch/job/storage/storage.go

```diff
@@ -117,3 +118,13 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 func (r *StatusREST) GetResetFields() map[fieldpath.APIVersion]*fieldpath.Set {
 	return r.store.GetResetFields()
 }
+
+func (r *StatusREST) Delete(ctx context.Context, name string, deleteValidation rest.ValidateObjectFunc, options *metav1.DeleteOptions) (runtime.Object, bool, error) {
+	if options.PropagationPolicy == nil {
```

Comment:
The other option is to have a method similar to WarningOnCreate:

```go
// WarningsOnDelete returns warnings for the deletion of job object. It can be moved to RESTDeleteStrategy interface but
// not sure, if there are any objects that can benefit from this movement. So, keeping this method local to
// jobStrategy for now.
func (jobStrategy) WarningsOnDelete(ctx context.Context, obj runtime.Object) []string {
	newJob := obj.(*batch.Job)

}
```

Reply:
I have a hard time seeing more than this use of a warning on delete, so I probably wouldn't add a general method like that yet

Reply:
```suggestion
	if options.PropagationPolicy == nil && options.OrphanDependents == nil {
```

Reply:
So, the approach is fine? Having set the warnings in storage? Because I see no other example of warnings being used in storage.

Reply:
And isn't OrphanDependents a deprecated one?

Reply:
> So, the approach is fine? Having set the warnings in storage? Because I see no other example of warnings being used in storage.

we tend to add warnings at the level that knows the relevant info... endpoint-level warnings for deprecated APIs are added in `k8s.io/apiserver/pkg/endpoints/*`, object-level warnings for create/update are added in strategy create and update methods, this is related to delete options the storage/strategy has an opinion about, so adding in delete implementations doesn't seem bad to me

> isn't OrphanDependents a deprecated one?

yes, but if the requester has explicitly set that, we don't have to wonder if they are getting implicit behavior they didn't intend